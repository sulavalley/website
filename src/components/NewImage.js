import React from 'react';
import propTypes from 'prop-types';
import base from '../base';
import { Button, Form } from 'semantic-ui-react';

class NewImage extends React.Component {
  state = {
    caption: '',
    src: ''
  };

  addImage() {
    base.push('images', {
      data: {
        caption: this.state.caption,
        src: this.state.src,
        thumbnail: this.state.src,
        thumbnailWidth: 320,
        thumbnailHeight: 212
      },
      then(err) {
        if (!err) {
          alert('La imagen ha sido aÃ±adida');
        } else {
          alert('Ha ocurrido un error');
        }
      }
    });
    this.setState({ caption: '', src: '' });
  }

  handleSubmit = event => {
    event.preventDefault();
    this.addImage();
  };

  render() {
    const { caption } = this.state;
    const { src } = this.state;

    return (
      <Form>
        <br />
        <Form.Field>
          <label>Nombre de la imagen</label>
          <input
            type="text"
            value={caption}
            placeholder="Nombre de la imagen"
            onChange={event => this.setState({ caption: event.target.value })}
          />
        </Form.Field>
        <Form.Field>
          <label>Imagen</label>
          <input
            type="url"
            value={src}
            placeholder="Enlace de la imagen"
            onChange={event => this.setState({ src: event.target.value })}
          />
        </Form.Field>
        <Button type="submit" onClick={this.handleSubmit}>
          Crear
        </Button>
      </Form>
    );
  }
}

NewImage.propTypes = {
  caption: propTypes.string.isRequired,
  src: propTypes.string.isRequired
};

export default NewImage;
