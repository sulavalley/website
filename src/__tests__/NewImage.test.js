import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NewImage from '../components/NewImage';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NewImage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<NewImage />);
});

test('renders correctly', () => {
  const tree = renderer.create(<NewImage />).toJSON();
  expect(tree).toMatchSnapshot();
});
