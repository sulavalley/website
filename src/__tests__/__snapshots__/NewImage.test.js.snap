// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  action={undefined}
  className="ui form"
  onSubmit={[Function]}
>
  <br />
  <div
    className="field"
  >
    <label>
      Nombre de la imagen
    </label>
    <input
      onChange={[Function]}
      placeholder="Nombre de la imagen"
      type="text"
      value=""
    />
  </div>
  <div
    className="field"
  >
    <label>
      Imagen
    </label>
    <input
      onChange={[Function]}
      placeholder="Enlace de la imagen"
      type="url"
      value=""
    />
  </div>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
    type="submit"
  >
    Crear
  </button>
</form>
`;
